### PR TITLE
RsMatchStatus matcher refactored to HmRsStatus

### PR DESCRIPTION
**Pull request by erimerturk**
RsMatchStatus matcher refactored to HmRsStatus. It is now encapsulating to another matcher #166

### CODE REVIEW

**Comment by davvd** [comment]
@erimerturk Many thanks, I will find someone to review it before we merge

**Comment by davvd** [comment]
@pinaf it's yours,please review

File: src/main/java/org/takes/facets/hamcrest/HmRsStatus.java [hunk 1]
```diff
@@ -41,17 +42,17 @@
 public final class HmRsStatus extends TypeSafeMatcher<Response> {
 
     /**
-     * Expected response status code for matcher.
+     * Expected response status code matcher.
      */
-    private final transient int expected;
+    private final transient Matcher<? extends Integer> expected;
```
Comment (pinaf) [inline src/main/java/org/takes/facets/hamcrest/HmRsStatus.java]: @erimerturk let's call this variable `matcher`

File: src/main/java/org/takes/facets/hamcrest/HmRsStatus.java [hunk 2]
```diff
@@ -41,17 +42,17 @@
...
 
     /**
-     * Expected input.
-     * @param input Is expected result code.
+     * Expected matcher.
+     * @param matcher Is expected result code matcher.
      */
-    public HmRsStatus(final int input) {
+    public HmRsStatus(final Matcher<? extends Integer> matcher) {
```
Comment (pinaf) [inline src/main/java/org/takes/facets/hamcrest/HmRsStatus.java]: @erimerturk and this parameter `mtchr`

File: src/main/java/org/takes/facets/hamcrest/HmRsStatus.java [hunk 3]
```diff
@@ -74,7 +74,7 @@ public boolean matchesSafely(final Response item) {
         try {
             final String head = item.head().iterator().next();
             final String[] parts = head.split(" ");
-            return this.expected == Integer.parseInt(parts[1]);
+            return this.expected.matches(Integer.parseInt(parts[1]));
         } catch (final IOException exception) {
```
Comment (pinaf) [inline src/main/java/org/takes/facets/hamcrest/HmRsStatus.java]: @erimerturk let's rename this `ex`

File: src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java [hunk 4]
```diff
@@ -49,11 +49,11 @@
     public void responseCodeIsEqualToExpected() throws IOException {
```
Comment (pinaf) [inline src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java]: @erimerturk we should rename this method `testsStatusOK`. Also, the javadoc for the method should start with `HmRsStatus can...`, so let's do `HmRsStatus can test status HTTP_OK`

File: src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java [hunk 5]
```diff
@@ -67,8 +67,8 @@ public void responseCodeIsNotEqualToExpected() throws IOException {
         MatcherAssert.assertThat(
             new TkHTML("<html><body/></html>").act(new RqFake()),
             Matchers.not(
-                Matchers.is(
```
Comment (pinaf) [inline src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java]: @erimerturk this method should be renamed to `testsStatusNotFound` and javadoc should read `HmRsStatus can test status HTTP_NOT_FOUND`

**Comment by pinaf** [comment]
@erimerturk 5 comments above.

**Comment by erimerturk** [comment]
@pinaf could you review CR fixes please

File: src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java [hunk 6]
```diff
@@ -42,33 +42,33 @@
...
         );
         MatcherAssert.assertThat(
             new TkEmpty().act(new RqFake()),
-            Matchers.is(new HmRsStatus(HttpURLConnection.HTTP_OK))
+            new HmRsStatus(Matchers.equalTo(HttpURLConnection.HTTP_OK))
         );
     }
 
     /**
      * Fail path tester.
```
Comment (pinaf) [inline src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java]: @erimerturk remove this line.

File: src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java [hunk 7]
```diff
@@ -42,33 +42,33 @@
...
-     * Should test expected code not equal case
+     * HmRsStatus can test status HTTP_NOT_FOUND
```
Comment (pinaf) [inline src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java]: @erimerturk let's add a dot at the end of the comment line.

File: src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java [hunk 8]
```diff
@@ -42,33 +42,33 @@
 
     /**
      * Happy path tester.
```
Comment (pinaf) [inline src/test/java/org/takes/facets/hamcrest/HmRsStatusTest.java]: @erimerturk remove this line.

**Comment by pinaf** [comment]
@erimerturk looks good. 3 minor comments and we are good to go.

**Comment by erimerturk** [comment]
@pinaf could you check and approve it please

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge pls

**Comment by davvd** [comment]
@elenavolokhova please, let us know what do you think about this ticket, according to our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by yegor256** [comment]
@rultor deploy

**Comment by elenavolokhova** [comment]
@pinaf 

According to our [quality rules](http://www.teamed.io/qa.html):
 > Comments were mostly about design problems, not cosmetic issues.
 
 There were mostly cosmetic issues found during review. Try to pay attention to bigger design problems and the overall quality of the code.
Please confirm that this new rule is clear.

**Comment by pinaf** [comment]
@elenavolokhova I disagree - variable names, method names and javadoc are part of the CR. Also, there were no design issues here.

**Comment by elenavolokhova** [comment]
@pinaf Agree. I applied this rule too formal. It is aimed to bring to light not effective reviews completed just to get "at least three problems in the code". This one is not one of them.

**Comment by elenavolokhova** [comment]
@davvd Quality is good here.

**Comment by davvd** [comment]
> @davvd Quality is good here.

@elenavolokhova thank you

**Comment by pinaf** [comment]
@elenavolokhova :)

**Comment by davvd** [comment]
@pinaf 10 mins was added to the account of @elenavolokhova (for QA review), in transaction 56855100

done, I added **19 mins** in payment `AP-3S9576173T967011S`, 56 hours and 48 mins spent to complete this

review comments (c=4) added as a bonus

added +19 to your rating, now it is equal to [+7672](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**